lib/sms/sms.c

```diff
@@ -173,19 +173,21 @@ static void sms_at_cmd_handler_cmt(const char *at_notif)
 		sms_buf_tmp);
 	if (err < 1) {
 		LOG_ERR("Unable to parse CMT notification, err=%d: %s", err, at_notif);
-		goto sms_ack_send;
+		goto error;
 	}
 
 	sms_data_info.type = SMS_TYPE_DELIVER;
 	err = sms_deliver_pdu_parse(sms_buf_tmp, &sms_data_info);
 	if (err) {
-		goto sms_ack_send;
+		goto error;
```

Comment:
I think the lib should send the ACK even before parsing the PDU.
I did it at the entry of this function.

Reply:
That's fine and simplifies the code nicely too although it shouldn't affect on the execution as this function will run first anyway before the ACK worker is executing.
I moved scheduling of the ACK to the beginning of the function.